src/smc-webapp/course/actions.ts

```diff
@@ -1455,11 +1456,10 @@ export class CourseActions extends Actions<CourseState> {
   //         .admin_upgrade_all_student_projects(cores:2)
   // The quotas are: cores, cpu_shares, disk_quota, memory, mintime, network, member_host
   public async admin_upgrade_all_student_projects(quotas): Promise<void> {
+    const account_store = this.redux.getStore("account");
     if (
-      !this.redux
-        .getStore("account")
-        .get("groups", [])
-        .contains("admin")
+      account_store &&
+      account_store.get("groups", { includes: _ => false }).includes("admin")
```

Comment:
This is frankly confusing to read.  Yes, I can read it, but really -- making an object with an includes method that returns false is kind of complicated.  I would prefer more straightforward code, e.g., 
```
const groups = account_store.get("groups")
if (groups != null && groups.includes("admin")) {
...
```

That's absolutely trivial to quickly read.  Whereas what you wrote above definitely slows a person down when reading it.  Harder to read code makes development slower, and we don't have a lot of money.

Reply:
Yeah, I'm being too clever here.